operators/hack/eck-dump.sh

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+# Diagnostics utility for Elastic Cloud in Kubernetes (ECK)
+
+set -eu
+
+help() {
+  echo 'Usage: ./eck-dump.sh [OPTIONS]
+
+Dumps Elastic Cloud in Kubernetes (ECK) info out suitable for debugging and diagnosing problems.
+
+By default, dumps everything to stdout. You can optionally specify a directory with --output-directory.
+By default only dumps things in the namespaces "elastic-system" and the current, but you can switch to
+different namespaces with the --operator-namespaces and --elastic-namespaces flags.
+
+Options:
+  -N, --operator-namespaces     Namespace(s) in which operator(s) are running in (list separated by comma)
+  -n, --resources-namespaces    Namespace(s) in which resources are managed      (list separated by comma)
+  -d, --output-directory        Path to output dump files (default: .)
+  -v, --verbose                 Verbose mode
+
+Dependencies:
+  kubectl'
+  exit
+}
+
+OPERATOR_NS=elastic-system
+ELASTIC_NS="default"
+OUTPUT_DIR=""
+VERBOSE=0
+
+parse_args() {
+  while :; do
+    local flag=${1:-""} value=${2:-""}
+    case $flag in
+    -h|--help)
+      help
+    ;;
+    -N|--operator-namespaces)
+      OPERATOR_NS=${value:-OPERATOR_NS}
+    ;;
+    -n|--elastic-namespaces)
+      ELASTIC_NS=${value:-${$(current_namespace):-ELASTIC_NS}}
+    ;;
+    -d|--output-directory)
+      OUTPUT_DIR=${value:-${OUTPUT_DIR:-"."}}
+    ;;
+    -v|--verbose)
+      VERBOSE=1
+    ;;
+    esac
+    shift || break
+  done
+}
+
+main() {
+  parse_args $@
+
+  IFS=, # use comma as field separator for iterations
+  
+  # start by checking the namespaces exist
+  local all_ns="$OPERATOR_NS,$ELASTIC_NS"
+  for ns in $all_ns; do
+    check_namespace $ns
+  done
+
+  # get global info
+  kubectl version   -o json | to_stdin_or_file version.json
+  kubectl get nodes -o json | to_stdin_or_file nodes.json
+
+  # get info from the namespaces in which operators are running in 
+  for ns in $OPERATOR_NS; do
+    get_resources $ns statefulsets
+    get_resources $ns pods
+    get_resources $ns services
+    get_resources $ns events
+    get_logs $ns
+  done
+
+  # get info from the namespaces in which resources are managed 
+  for ns in $ELASTIC_NS; do
+    get_resources $ns replicasets
+    get_resources $ns deployments
+    get_resources $ns pods
+    get_resources $ns services
+    get_resources $ns events
+    
```

Comment:
I think we also want to list secrets and configmaps (but maybe not their content since that can include other apps and would be fairly intrusive)?
Also retrieve PVCs and PVs.

Reply:
I added pv,pvc,cm and secrets without content.